Show count of enabled payment methods in main settings page

Mayisha opened the pull request:
Fixes #1726

### Description
When UPE preview and UPE checkout feature flags are enabled, currently the method name is `Stripe UPE`. Changes in this PR show the number of enabled payment methods beside the method name. This change is only applicable to the settings page only, not the checkout page.

**When `_wcstripe_feature_upe_settings` is enabled**
For 1 method enabled
![1 method](https://user-images.githubusercontent.com/33387139/133218861-4399f514-941b-452e-822d-c970404741e2.png)

For multiple methods enabled
![4 methods](https://user-images.githubusercontent.com/33387139/133218865-a73950a3-9c87-4a67-a3a6-7bbad19f715e.png)

**When `_wcstripe_feature_upe_settings` is disabled**
![without ff](https://user-images.githubusercontent.com/33387139/133218907-a17c8284-e1ee-4f75-8223-5e04fe92a514.png)

### Testing instructions
- Enable UPE preview and UPE checkout

### QA checklist
- [ ] Count of enabled payment method is visible beside the method name on the settings page
- [ ] Count of enabled payment method is not visible on the checkout page
- [ ] UI is unchanged when the feature flag is disabled

Mayisha committed:
Show count of enabled payment methods in main settings page

Mayisha commented:
### Note
- Design shows that we have count of methods and icons of methods beside the method name. Confirmed with @jarekmorawski in [this slack discussion](https://a8c.slack.com/archives/C01RTUVR81K/p1631551019078200) that we are going to show count only. Previously I had created new React component and enqueued scripts to show the count and icons. But as we need to show the count only, not using the React component anymore, rather added the count in the `title` field of the row.
- Please confirm I am using the right feature flag to add this feature. The issue says `Description is updated only when the UPE flag is enabled`. But the new method name `Stripe UPE` is shown when `_wcstripe_feature_upe_settings` so added the count as when this flag is enabled.

EvanDotPro commented:
I don't believe this should (or does) depend on the `_wcstripe_feature_upe_settings` option/flag? :thinking:

Mayisha commented:
> I don't believe this should (or does) depend on the `_wcstripe_feature_upe_settings` option/flag?

@EvanDotPro do you mean the count should be there always?

EvanDotPro commented:
> do you mean the count should be there always?

@Mayisha The count should be displayed when `_wcstripe_feature_upe` and `woocommerce_stripe_settings`.`upe_checkout_experience_enabled` are both set to 'yes' (`WC_Stripe_Feature_Flags::is_upe_preview_enabled()` and `WC_Stripe_Feature_Flags::is_upe_checkout_enabled()` respectively).

Looking at the diff here, that should already be happening automatically since the change is in `WC_Stripe_UPE_Payment_Gateway` which is only loaded when both of those options are set to 'yes'.

EvanDotPro commented:
The `_wcstripe_feature_upe_settings` flag only indicates whether the new settings UI is being used or not. There's a chance UPE may get released initially without the new settings UI, so the count should show when UPE is enabled, even if the old (current) settings UI is still being used.

From what I can tell, your PR as it is now, behaves as I'm describing — no changes needed.

frosso commented on includes/payment-methods/class-wc-stripe-upe-payment-gateway.php:
```diff
@@ -96,14 +96,15 @@ public function __construct() {
-		$this->statement_descriptor = ! empty( $main_settings['statement_descriptor'] ) ? $main_settings['statement_descriptor'] : '';
+		$main_settings                 = get_option( 'woocommerce_stripe_settings' );
+		$enabled_payment_methods_count = count( $this->get_upe_enabled_payment_method_ids() );
+		$this->title                   = $enabled_payment_methods_count . ( $enabled_payment_methods_count > 1 ? ' payment methods' : ' payment method' );
```
Just giving this a quick look - can you confirm that the payment method _doesn't_ show this message no the checkout page? I _believe_ this property is also used for the checkout page, when UPE is enabled.

Mayisha replied:
It is actually showing up there :woman_facepalming: 
I will fix this.

harriswong commented:
👀

Mayisha committed:
Update logic for showing methods count in title
- Show count only when Upe checkout and Upe preview is enabled
- Show only in settings page, not in the checkout page.

Mayisha commented:
Fixed the condition for showing the method count in the title. Now it shows only on the settings page when UPE preview and UPE checkout flags are enabled. The checkout page remains unchanged.

frosso commented on includes/payment-methods/class-wc-stripe-upe-payment-gateway.php:
```diff
@@ -112,6 +112,12 @@ public function __construct() {
+		// When feature flags are enabled, title shows the count of enabled payment methods in settings page only.
+		if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() && WC_Stripe_Feature_Flags::is_upe_preview_enabled() && isset( $_GET['page'] ) && 'wc-settings' === $_GET['page'] ) {
+			$enabled_payment_methods_count = count( $this->get_upe_enabled_payment_method_ids() );
+			$this->title                   = $enabled_payment_methods_count . ( $enabled_payment_methods_count > 1 ? ' payment methods' : ' payment method' );
```
This should probably be wrapped in a translation function (`_n(...)` - https://codex.wordpress.org/I18n_for_WordPress_Developers#Plurals)

Mayisha replied:
Thank you. Addressed :tada:

Mayisha committed:
handle plural form with _n()

frosso reviewed (approved):
Looks good, thanks for the changes! I'm unsure about the scenario where there's 0 methods enabled, but we can always change it!

Mayisha committed:
Merge branch 'develop' into add/1726-enabled-methods-count
